native/src/main/java/io/ballerina/persist/cmd/Push.java

```diff
@@ -0,0 +1,257 @@
+        Connection connection;
+        Statement statement;
+
+        if (helpFlag) {
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(COMMAND_IDENTIFIER);
+            errStream.println(commandUsageInfo);
+            return;
+        }
+
+        boolean isTest = (projectEnvironmentBuilder != null);
+        try  {
+            URL[] urls = {};
+            JdbcDriverLoader driverLoader;
+            if (!isTest) {
+                balProject = ProjectLoader.loadProject(Paths.get(""));
+            } else {
+                balProject = ProjectLoader.loadProject(Paths.get(sourcePath), projectEnvironmentBuilder);
+            }
+            driverLoader = new JdbcDriverLoader(urls, driverPath.toAbsolutePath());
+            Class drvClass = driverLoader.loadClass(MYSQL_CLASS);
+            driver = (Driver) drvClass.getDeclaredConstructor().newInstance();
+            name = balProject.currentPackage().descriptor().org().value() + "." + balProject.currentPackage()
+                    .descriptor().name().value();
+        } catch (ProjectException e) {
+            errStream.println("The current directory is not a Ballerina project!");
+            return;
+        } catch (ClassNotFoundException e) {
+            errStream.println("Driver Not Found");
+            return;
+        } catch (InstantiationException | InvocationTargetException e) {
+            errStream.println("Error instantiation the jdbc driver");
+            return;
+        } catch (IllegalAccessException e) {
+            errStream.println("Access denied trying to instantiation the jdbc driver");
+            return;
+        } catch (NoSuchMethodException e) {
+            errStream.println("Method not fount error while trying to instantiate jdbc driver : " + e.getMessage());
+            return;
+        } catch (MalformedURLException e) {
+            errStream.println("Error in jdbc driver path : " + e.getMessage());
+            return;
+        }
+        try {
+            if (projectEnvironmentBuilder == null) {
+                balProject = BuildProject.load(Paths.get(sourcePath).toAbsolutePath());
+            } else {
+                balProject = BuildProject.load(projectEnvironmentBuilder, Paths.get(sourcePath).toAbsolutePath());
+            }
+        } catch (ProjectException e) {
+            errStream.println(e.getMessage());
+            return;
+        }
+        try {
+            balProject.currentPackage().getCompilation();
+        } catch (ProjectException e) {
+            errStream.println(e.getMessage());
+            return;
+        }
+        try {
+            configurations = SyntaxTreeGenerator.readToml(
+                    Paths.get(this.sourcePath, this.configPath), name);
+
+            Path path = Paths.get(this.sourcePath, TARGET, SQL_PATH);
+
+            FileReader fileReader = new FileReader(path.toAbsolutePath().toString());
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
```

Comment:
we need to close the reader even if there is an error. shall we move the reader initialization code in to `try-with-resource`? or close the reader inside finally block